lib/RestructureCFG/RestructureCFG.cpp

```diff
@@ -1492,16 +1493,23 @@ bool restructureCFG(Function &F, ASTTree &AST) {
 
     float Increase = float(FinalWeight) / float(InitialWeight);
 
-    std::ofstream Output;
+    ExitOnError ExitOnError;
+    std::error_code EC;
+
     const char *FunctionName = F.getName().data();
-    std::ostream &OutputStream = pathToStream(MetricsOutputPath + "/"
-                                                + FunctionName,
-                                              Output);
-    OutputStream << "function,"
-                    "duplications,percentage,tuntangle,puntangle,iweight\n";
-    OutputStream << F.getName().data() << "," << DuplicationCounter << ","
-                 << Increase << "," << UntangleTentativeCounter << ","
-                 << UntanglePerformedCounter << "," << InitialWeight << "\n";
+    llvm::ToolOutputFile OutputFile(MetricsOutputPath + "/" + FunctionName,
+                                    EC,
+                                    llvm::sys::fs::OF_Text);
+    if (EC)
+      ExitOnError(llvm::createStringError(EC, EC.message()));
```

Comment:
`ExitOnError` is OK only on `Main.cpp`. Here, using the message.

Reply:
It's not clear what you mean with `Here, using the message`. Do you want me to `revng_abort` with `EC.message`?

Reply:
Yes

Reply:
I suspected so it's already done